From: Juergen Borleis <jbe@pengutronix.de>
Date: Thu, 3 Dec 2015 16:49:24 +0100
Subject: [PATCH] DTS/network: describe the network/switch setup

Keep in mind the hardware setup to be not confused by wrongly connected
phys to eth0. There is no! Keep the Micrel driver away, as it does the
wrong things, after the FEC driver has autodetected three phys...

Signed-off-by: Juergen Borleis <jbe@pengutronix.de>
---
 arch/arm/boot/dts/imx6qdl-cpu.dtsi  | 17 +++++++++++++++++
 arch/arm/boot/dts/imx6qdl-cpuA.dtsi |  8 --------
 2 files changed, 17 insertions(+), 8 deletions(-)

diff --git a/arch/arm/boot/dts/imx6qdl-cpu.dtsi b/arch/arm/boot/dts/imx6qdl-cpu.dtsi
index 6ad8cd2dda7f..373f524f46e8 100644
--- a/arch/arm/boot/dts/imx6qdl-cpu.dtsi
+++ b/arch/arm/boot/dts/imx6qdl-cpu.dtsi
@@ -302,12 +302,29 @@
 	phy-supply = <&reg_3p3v>;
 	#address-cells = <0>;
 	#size-cells = <1>;
+
+	/*
+	 * MDIO phy#1 is port 1 e.g. LAN1
+	 * MDIO phy#2 is port 2 e.g. LAN2
+	 * MDIO phy#2 is port 2 e.g. RMII port, with no real and usable
+	 * phy emulation
+	 */
 	fixed-link {
 		speed = <100>;
 		full-duplex;
 	};
 };
 
+/* signal "active link" on the RMII to the ethernet switch */
+&gpio1 {
+	active_rmii_link {
+		gpio-hog;
+		gpios = <&gpio1 24 GPIO_ACTIVE_LOW>;
+		output-low;
+		line-name = "rmii-active-link";
+	};
+};
+
 &i2c2 {
 	status = "okay";
 	clock-frequency = <100000>;
diff --git a/arch/arm/boot/dts/imx6qdl-cpuA.dtsi b/arch/arm/boot/dts/imx6qdl-cpuA.dtsi
index aa8a2e7bfbd2..327de3581f14 100644
--- a/arch/arm/boot/dts/imx6qdl-cpuA.dtsi
+++ b/arch/arm/boot/dts/imx6qdl-cpuA.dtsi
@@ -19,14 +19,6 @@
 	interrupts = <14 IRQ_TYPE_LEVEL_LOW>;
 };
 
-/* MX6QDL_PAD_ENET_RX_ER__GPIO1_IO24 is a gpio which should be high */
-&gpio1 {
-	status = "okay";
-	active_status = "high";
-	gpios = <&gpio1>;
-	pin = <24>;
-};
-
 &pinctrl_board {
 	pinctrl_ads7846: ads7846grp {
 		fsl,pins = <
